.github/workflows/build-artifact-s3.yml

```diff
@@ -35,6 +35,10 @@ jobs:
       with:
         submodules: true
         fetch-depth: 0
+    - uses: Swatinem/rust-cache@v2
```

Comment:
this may not do anything since the build happens in a docker container for the `messense/maturin-action` step

We should check if the `target` dir is volume mounted

Reply:
Looks like we should prob use sccache instead

https://github.com/PyO3/maturin-action/issues/53

Reply:
You can try turning on the `sccache` option for the `messense/maturin-action` step seen here:

https://github.com/PyO3/maturin-action

Reply:
Oh lol yeah I was literally looking at this as you posted and saw the notification come up on the maturin-action issue 😂 . Going to try `sccache`